src/aioquic/quic/connection.py

```diff
@@ -1147,6 +1162,17 @@ def _find_network_path(self, addr: NetworkAddress) -> QuicNetworkPath:
         self._logger.debug("Network path %s discovered", network_path.addr)
         return network_path
 
+    def _find_network_challenge(self) -> QuicNetworkPathChallenge:
+        # check existing network challenge
+        for idx, network_chall in enumerate(self._network_challenges):
```

Comment:
This looks like a complicated way of saying "always take the last challenge in the list"?